lucille-core/src/main/java/com/kmwllc/lucille/stage/DictionaryLookup.java

```diff
@@ -94,44 +116,61 @@ private HashMap<String, String[]> buildHashMap(String dictPath) throws StageExce
           continue;
         }
 
+        // save the first word for both single and multi-word lines
+        String word = line[0].trim();
+
         // TODO : Add log messages for when encoding errors occur so that they can be fixed
+        String[] value;
         if (line.length == 1) {
-          String word = line[0].trim();
-          if (ignoreCase) {
-            dict.put(word.toLowerCase(), new String[]{word});
-          } else {
-            dict.put(word, new String[]{word});
-          }
+          value = setOnly ? PRESENT : new String[]{word};
+        } else if (setOnly) {
+          LOG.warn(String.format("Entry \"%s\" on line %d contained payloads which were ignored. " +
+              "This dictionary entry will be treated as a set.", word, reader.getLinesRead()));
+          value = PRESENT;
         } else {
           // Handle multiple payload values here.
-          String[] rest = Arrays.copyOfRange(line, 1, line.length);
-          for (int i = 0; i < rest.length; i++) {
-            rest[i] = rest[i].trim();
-          }
-          if (ignoreCase) {
-            dict.put(line[0].trim().toLowerCase(), rest);
-          } else {
-            dict.put(line[0].trim(), rest);
-          }
+          value = Arrays.stream(Arrays.copyOfRange(line, 1, line.length)).map(String::trim).toArray(String[]::new);
         }
+        // Add the word and its payload(s) to the dictionary
+        dict.put(ignoreCase ? word.toLowerCase() : word, value);
       }
     } catch (IOException e) {
       throw new StageException("Failed to read from the given file.", e);
     } catch (CsvValidationException e) {
       throw new StageException("Error validating CSV", e);
     }
-
     return dict;
   }
 
   @Override
   public Iterator<Document> processDocument(Document doc) throws StageException {
+
+    // todo consider what should happen if using set and destination field is already populated
+    if (setOnly) {
+      for (String destField : destFields) {
+        doc.removeField(destField);
+      }
+    }
+
```

Comment:
Is it okay to remove fields that are already present in the document for this use case? I am considering adding a warning or documentation that a field is being removed.

Reply:
Let's try to keep the behavior of this stage as consistent as possible whether or not setOnly is true or false. When setOnly==false we don't delete the destFields. We use doc.update along with the provided updateMode, which allows the user to control whether they want replace or append behavior -- see: `doc.update(destField, updateMode, outputValues.toArray(new String[0]))`.  Let's use the same approach when setOnly==true -- don't delete destFields at the beginning. Do the update via doc.update()